programs/marinade-finance/src/state/fee.rs

```diff
@@ -60,9 +59,66 @@ impl FromStr for Fee {
     type Err = Error; // TODO: better error
 
     fn from_str(s: &str) -> Result<Self> {
-        f64::try_into(
-            s.parse()
-                .map_err(|_| error!(MarinadeError::CalculationFailure))?,
-        )
+        f64::try_into(s.parse()?)
+    }
+}
+
+/// FeeCents, same as Fee but / 1_000_000 instead of 10_000
+/// 1 FeeCent = 0.0001%, 10_000 FeeCent = 1%, 1_000_000 FeeCent = 100%
+#[derive(
+    Clone, Copy, Debug, Default, AnchorSerialize, AnchorDeserialize, PartialEq, Eq, PartialOrd, Ord,
+)]
+pub struct FeeCents {
+    pub bp_cents: u32,
+}
+
+impl Display for FeeCents {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // use integer division to avoid including f64 libs
+        write!(f, "{}.{}%", self.bp_cents / 10000, self.bp_cents % 10000)
```

Comment:
Please set 0 padding for the second {} to not have 5.1% printed for 50001 bp-cents